File: code-dist-2019-10/Python/rei2p6/basis_pursuit_lp.py
import cvxpy as cp
import numpy as np
m, n = 5, 10
np.random.seed(1)
A, b = np.random.randn(m,n), np.random.randn(m)
x, z = cp.Variable(n), cp.Variable(n)
obj  = cp.Minimize( np.ones(n) * z )
cons = [A * x == b,
        z >=  x,
        z >= -x]
P = cp.Problem(obj, cons)
P.solve(verbose=True)
print(x.value)

File: code-dist-2019-10/MATLAB/rei2p6/basis_pursuit_lp.m
m = 5;  n = 10;
rng(0,'twister');
A = randn(m,n);  b = randn(m,1);
cvx_begin
    variables x(n,1) z(n,1)
    minimize( ones(1,n) * z )
    subject to
        A * x == b;
        z >=  x;
        z >= -x;
cvx_end